activesupport/test/option_merger_test.rb

```diff
@@ -114,6 +114,17 @@ def test_option_merger_implicit_receiver
     assert_equal expected, @options
   end
 
+  def test_with_options_hash_like
+    hash_like = Struct.new :hash do
+      delegate_missing_to :hash
```

Comment:
Is that really a Hash-like?

The method that matter here is `to_hash`, that's probably the only method a good mock should respond to. Because here you respond to anything, so if `OptionMerger` was modified to call `to_h`, this test would still pass, which I don't think is good.

Reply:
@byroot thank you for the review! 

> The method that matter here is `to_hash`

I've removed the `delegate_missing_to` and replaced it with a more explicit `delegate` call.

The current implementation also incorporates a check for `deep_merge`, so I've added that to the list too.